models/repo/archive_download_count.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package repo
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/git"
+	api "code.gitea.io/gitea/modules/structs"
+)
+
+// RepoArchiveDownloadCount counts all archive downloads for a tag
+type RepoArchiveDownloadCount struct {
+	ID     int64           `xorm:"pk autoincr"`
+	RepoID int64           `xorm:"index unique(s)"`
+	Type   git.ArchiveType `xorm:"unique(s)"`
+	Tag    string          `xorm:"index unique(s)"`
+	Count  int64
+}
+
+func init() {
+	db.RegisterModel(new(RepoArchiveDownloadCount))
+}
+
+// CountArchiveDownload adds one download the the given archive
+func CountArchiveDownload(ctx context.Context, repoID int64, tp git.ArchiveType, tag string) error {
```

Comment:
`db.GetEngine(ctx).Incr(...)`

Reply:
Does this also insert a row when not found?